extension/src/popup/components/sendPayment/SendSettings/TransactionFee/index.tsx

```diff
@@ -1,39 +1,135 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Formik, Form, Field, FieldProps } from "formik";
+import BigNumber from "bignumber.js";
 
-import { Button, Input } from "@stellar/design-system";
+import StellarSdk from "stellar-sdk";
+import {
+  Button,
+  Input,
+  Icon,
+  TextLink,
+  DetailsTooltip,
+} from "@stellar/design-system";
 
 import { navigateTo } from "popup/helpers/navigate";
 import { ROUTES } from "popup/constants/routes";
-
 import { PopupWrapper } from "popup/basics/PopupWrapper";
+import { FormRows } from "popup/basics/Forms";
+import {
+  saveTransactionFee,
+  transactionDataSelector,
+} from "popup/ducks/transactionSubmission";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
 
 import "./styles.scss";
 
-export const SendSettingsFee = ({
-  transactionFee,
-  setTransactionFee,
-}: {
-  transactionFee: string;
-  setTransactionFee: (state: string) => void;
-}) => (
-  <PopupWrapper>
-    <div className="TransactionFee">
-      <div className="TransactionFee__header">Transaction Fee</div>
-      <Input
-        id="transaction-fee-input"
-        className="SendTo__input"
-        value={transactionFee}
-        onChange={(e: React.ChangeEvent<any>) =>
-          setTransactionFee(e.target.value)
+enum NetworkCongestion {
+  LOW = "Low",
+  MEDIUM = "Medium",
+  HIGH = "High",
+}
+
+const lumensFromStroops = (stroops: BigNumber | string): BigNumber => {
```

Comment:
there's actually a helper for this in extension/src/helpers/stellar.ts called `stroopToXlm`

Reply:
ahh nice. I also changed it to use BigNumber in case values have large decimals
https://github.com/stellar/freighter/pull/401/commits/53c40fcd27a84e5d590fae0263f3922082d00ae7